src/compile/model.ts

```diff
@@ -361,6 +361,20 @@ export abstract class Model {
   public getComponent(type: 'scales' | 'selection', name: string): any {
     return this.component[type][name] || this.parent.getComponent(type, name);
   }
+
+  public getLookups(predicate?: (lookup: LookupTransform) => boolean): LookupTransform[] {
+    const lookups: LookupTransform[] = this.transforms.filter((t: Transform): t is LookupTransform => isLookup(t) && (!predicate || predicate(t)));
+    return lookups;
+  }
+}
+
+// above works if we add this, though it should be already added from typescript!!!
+// See this PR https://github.com/Microsoft/TypeScript/issues/7657 which was merged in ~ TS 2.1
+// TODO: remove this before merge. If we can't figure out TS, just have getLookups return Transform[]
+declare global {
```

Comment:
this is DEFINITELY not staying. it's weird that its even needed since the same line is in Typescript's lib.d.ts and therefore the above filter should throw an error like it does if we remove this global bit. So weird.